src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/LeoAuthProviderHelper.scala

```diff
@@ -32,48 +35,51 @@ object LeoAuthProviderHelper {
   }
 }
 
-class LeoAuthProviderHelper(wrappedAuthProvider: LeoAuthProvider, authConfig: Config, serviceAccountProvider: ServiceAccountProvider) extends LeoAuthProvider(authConfig, serviceAccountProvider) with LazyLogging {
+class LeoAuthProviderHelper(val wrappedProvider: LeoAuthProvider, authConfig: Config, serviceAccountProvider: ServiceAccountProvider)(implicit val system: ActorSystem)
+  extends LeoAuthProvider(authConfig, serviceAccountProvider) with SamProviderHelper[LeoAuthProvider] {
```

Comment:
I might be misreading, but:

1. I think it should be down to the provider whether it caches or not.
2. Isn't adding `with SamProviderHelper` here a bit too high in the hierarchy? You don't know if the SA provider is a Sam one. It might not be.

Reply:
2. You are right, thanks. We shouldn't be assuming Sam here. I think I abstracted a bit too much.

Reply:
1. Yeah. So maybe the caching should be in our Sam provider, not the helper